domain/wollemi/service_format_test.go

```diff
@@ -226,6 +226,97 @@ func (t *ServiceSuite) TestService_GoFormat() {
 
 		require.NoError(t, wollemi.GoFormat(rewrite, []string{"app/server"}))
 	})
+
+	t.It("can resolve dependencies by import_path", func(t *T) {
+		data := t.GoFormatTestData()
+
+		delete(data.Stat, "third_party/go/github.com/spf13/BUILD.plz")
+		delete(data.Lstat, "third_party/go/github.com/spf13/BUILD.plz")
+		delete(data.Parse, "third_party/go/github.com/spf13/BUILD.plz")
+
+		data.Stat["third_party/go/github.com/spf13/cobra/BUILD.plz"] = &FileInfo{
+			FileName: "BUILD.plz",
+			FileMode: os.FileMode(420),
+		}
+		data.Lstat["third_party/go/github.com/spf13/cobra/BUILD.plz"] = data.Stat["third_party/go/github.com/spf13/cobra/BUILD.plz"]
+
+		data.Stat["third_party/go/github.com/spf13/pflag/BUILD.plz"] = &FileInfo{
+			FileName: "BUILD.plz",
+			FileMode: os.FileMode(420),
+		}
+		data.Lstat["third_party/go/github.com/spf13/pflag/BUILD.plz"] = data.Stat["third_party/go/github.com/spf13/pflag/BUILD.plz"]
+
+		data.Walk = make([]string, 0, len(data.Walk))
+		for path, _ := range data.Lstat {
+			data.Walk = append(data.Walk, path)
+		}
+
+		sort.Strings(data.Walk)
+
+		data.Parse["third_party/go/github.com/spf13/cobra/BUILD.plz"] = &please.BuildFile{
+			Path: "third_party/go/github.com/spf13/cobra/BUILD.plz",
+			Stmt: []please.Expr{
+				please.NewCallExpr("go_library", []please.Expr{
+					please.NewAssignExpr("=", "name", "cobra"),
+					please.NewAssignExpr("=", "srcs", please.NewGlob([]string{"*.go"}, "*_test.go")),
+					please.NewAssignExpr("=", "import_path", "github.com/spf13/cobra"),
+					please.NewAssignExpr("=", "deps", []string{"//third_party/go/github.com/spf13/pflag"}),
+					please.NewAssignExpr("=", "visibility", []string{"PUBLIC"}),
+				}),
+			},
+		}
```

Comment:
There are all sorts of requirements, of course, for real build rules in please. For example, a `go_library` should have `srcs` and be visible to other rules which depend on it. However these types of requirements don't really matter in the context of wollemi. Imho it's much better to define the bare minimum that is required for tests. It really helps me see the specific thing being tested and not get lost in the details.

```golang
cobra := "third_party/go/github.com/spf13/cobra/BUILD.plz"
data.Stat[cobra] = &FileInfo{
	FileName: "BUILD.plz",
	FileMode: os.FileMode(420),
}
data.Lstat[cobra] = data.Stat[cobra]

data.Walk = make([]string, 0, len(data.Walk))
for path, _ := range data.Lstat {
	data.Walk = append(data.Walk, path)
}

sort.Strings(data.Walk)

data.Parse[cobra] = &please.BuildFile{
	Path: cobra,
	Stmt: []please.Expr{
		please.NewCallExpr("go_library", []please.Expr{
			please.NewAssignExpr("=", "name", "cobra"),
			please.NewAssignExpr("=", "import_path", "github.com/spf13/cobra"),
		}),
	},
}
```